### ISSUE TITLE
bug: Failed to send the request: Upload Id not found in create-multipart-upload response for Upload without known content-length

### ISSUE THREAD

**Issue by jonathanswenson** [localstack/localstack#8860]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Using the s3 CRT library to perform a multi-part upload with an unknown file size leads to the upload reporting failure due to a missing `Upload ID` from the create-multipart-upload response. 

Upload request fails with the below error after 10 second (timeout) 
```
software.amazon.awssdk.core.exception.SdkClientException: Failed to send the request: Upload Id not found in create-multipart-upload response
java.util.concurrent.CompletionException: software.amazon.awssdk.core.exception.SdkClientException: Failed to send the request: Upload Id not found in create-multipart-upload response
	at software.amazon.awssdk.utils.CompletableFutureUtils.errorAsCompletionException(CompletableFutureUtils.java:65)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncExecutionFailureExceptionReportingStage.lambda$execute$0(AsyncExecutionFailureExceptionReportingStage.java:51)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at software.amazon.awssdk.utils.CompletableFutureUtils.lambda$forwardExceptionTo$0(CompletableFutureUtils.java:79)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncRetryableStage$RetryingExecutor.maybeAttemptExecute(AsyncRetryableStage.java:103)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncRetryableStage$RetryingExecutor.maybeRetryExecute(AsyncRetryableStage.java:184)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncRetryableStage$RetryingExecutor.lambda$attemptExecute$1(AsyncRetryableStage.java:159)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at software.amazon.awssdk.utils.CompletableFutureUtils.lambda$forwardExceptionTo$0(CompletableFutureUtils.java:79)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeAsyncHttpRequestStage.lambda$null$0(MakeAsyncHttpRequestStage.java:103)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeAsyncHttpRequestStage.completeResponseFuture(MakeAsyncHttpRequestStage.java:240)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeAsyncHttpRequestStage.lambda$executeHttpRequest$3(MakeAsyncHttpRequestStage.java:163)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: software.amazon.awssdk.core.exception.SdkClientException: Failed to send the request: Upload Id not found in create-multipart-upload response
	at app//software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:111)
	at app//software.amazon.awssdk.core.exception.SdkClientException.create(SdkClientException.java:43)
	at app//software.amazon.awssdk.services.s3.internal.crt.S3CrtResponseHandlerAdapter.handleError(S3CrtResponseHandlerAdapter.java:128)
	at app//software.amazon.awssdk.services.s3.internal.crt.S3CrtResponseHandlerAdapter.onFinished(S3CrtResponseHandlerAdapter.java:94)
	at app//software.amazon.awssdk.crt.s3.S3MetaRequestResponseHandlerNativeAdapter.onFinished(S3MetaRequestResponseHandlerNativeAdapter.java:24)
```

### Expected Behavior

Upload completes successfully. (which is what happens against S3 in AWS)

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
from docker-compose file
```
  localstack-community:
    profiles:
      - localstack-community
    container_name: localstack_main_community
    image: localstack/localstack
    ports:
      - '127.0.0.1:4566:4566' # LocalStack Gateway
      - '127.0.0.1:4510-4559:4510-4559' # external services port range
    environment:
      - DEBUG=${LOCALSTACK_DEBUG-}
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - 'localstack-community:/var/lib/localstack'
      - '/var/run/docker.sock:/var/run/docker.sock'
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Library versions (from gradle file): 
```
    implementation(platform("software.amazon.awssdk:bom:2.20.118"))
    implementation("software.amazon.awssdk.crt:aws-crt:0.24.0")
    implementation("software.amazon.awssdk:s3-transfer-manager")
    implementation("software.amazon.awssdk:s3")
    implementation("software.amazon.awssdk:sso")

    // https://mvnrepository.com/artifact/commons-codec/commons-codec
    implementation("commons-codec:commons-codec:1.16.0")
```

```java
  @Test
  public void multipartUploadWithUnknownSize() throws Exception {
    S3AsyncClient s3AsyncClient = S3AsyncClient.crtBuilder()
      .credentialsProvider(ProfileCredentialsProvider.create(profileName))
      .endpointOverride(new URI("http://localhost:4566"))
      .forcePathStyle(true)
      .region(Region.US_WEST_2)
      .build();

    try (S3TransferManager transferManager = S3TransferManager.builder().s3Client(s3AsyncClient).build()) {
      String key = UUID.randomUUID().toString();
      System.out.println("key is: "+ key);

      // unknown content length (does not happen with a known content length).
      BlockingOutputStreamAsyncRequestBody body = AsyncRequestBody.forBlockingOutputStream(null);

      UploadRequest uploadRequest = UploadRequest.builder()
        .putObjectRequest(PutObjectRequest.builder().bucket(bucket).key(key).build())
        .requestBody(body)
        .build();

      Upload uploadResponse = transferManager.upload(uploadRequest);

      try (OutputStream outputStream = body.outputStream()) {
        outputStream.write("Hello, world!\n".getBytes(StandardCharsets.UTF_8));
      } catch (IllegalStateException e) {
        // ignore this exception (it's expected here). The outputStream call will time out after 10 seconds
        // because the upload request fails due to the bad response.
      }
      // throws error 
      uploadResponse.completionFuture().join();
    }
  }
```

### Environment

```markdown
- OS: Mac OSX 13.4.1 (M1)
- LocalStack: 2.0.0.dev
```

### Anything else?

Localstack logs show a 200 OK response to the CreateMultipartUpload request. 

```
2023-08-08 09:51:14 2023-08-08T16:51:14.342  INFO --- [   asgi_gw_5] localstack.request.aws     : AWS s3.CreateMultipartUpload => 200
```

**Comment by bentsku** [comment]
Hello @jonathanswenson and thanks for your report! I'm noticing you're using an older image of LocalStack. I've just ran your test (thanks a lot for the nice reproduction sample!) with LocalStack 2.2.0 and the test ran successfully.

Could you give it a try with a newer version (`2.2.0` or `latest`) and see if that solves your issue? Thanks!

**Comment by jonathanswenson** [comment]
Yep looks fixed in 2.2.0. Sorry about that. 

I had even pulled down master and run from what I thought was source, but it turns out that I was still running against localstack 2.0.0 (despite using localstack cli 2.2.0.dev)
```
 💻 LocalStack CLI 2.2.1.dev

[16:13:43] starting LocalStack in Docker mode 🐳                                                                                                                                                                                                                             ...
LocalStack version: 2.0.0.dev
LocalStack Docker container id: 03f77ea2ec63
LocalStack build date: 2023-03-25
LocalStack build git hash: 739a5965
```

Pulling the container with 2.2.0 (just a change to the docker compose file) seems to have fixed the issue. Thanks!

**Comment by bentsku** [comment]
Awesome, thanks for the feedback! Sorry for the inconvenience on your side. I'm glad this is fixed!
